.github/workflows/main.yml

```diff
@@ -0,0 +1,69 @@
+# This is a basic workflow to help you get started with Actions
+
+name: Poliastro Website CI
+
+# Controls when the action will run. 
+on:
+  # Triggers the workflow on push or pull request events but only for the sources branch
+  push:
+    branches: 
+      - sources
+      - 'content'
+  
+  pull_request:
+    branches: 
+      - sources
+      - 'content'
+
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+jobs:
+  # Builds the WebPages locally before Deployment
+  build:
+    # Runner Type(s)
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: [ 3.7, 3.8, 3.9]
+
+    # Steps represent a sequence of tasks that will be executed as part of the job
+    steps:
+      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
+      - uses: actions/checkout@v2
+      # Set up Python Environment based on the version
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+      # Installations for all the dependencies 
+      - name: Install dependencies
+        run: |
+            python -m pip install --upgrade pip
+            pip install -r requirements.in
+      # Add the support plugins required & builds the webpages in HTML format
+      - name: Build the webpages
+        run: |
+            git submodule update --init --recursive
+            make html
+
+    # Deployment to GH Pages
+  deploy:
+    # Runner Type(s)
+    runs-on: ubuntu-latest
+    # Dependents on Build 
+    needs: build
+    strategy:
+      matrix:
+        python-version: [3.9]
+
+    steps:
+      - name: Displaying Python Versions
+        run: python -c "import sys; print(sys.version)"
+
+    # # Uncomment for deployment
```

Comment:
My suggestion is that you try this on your fork, to make sure it works. If you did already, could you please send a link to the relevant pipelines and their results?

Reply:
Sure, I have tried it on my fork & have successfully triggered the pipeline execution via GitHub Actions ... the link for the same is [here](https://github.com/DhruvSondhi/poliastro.github.io/actions) ... Please check the same :) 

*(PS: I have left the `deploy` job for configuration ... please make the necessary changes so as to meet the requirements 😄)*

Reply:
The most recent run for the pipeline can be found [here](https://github.com/DhruvSondhi/poliastro.github.io/actions/runs/594940098)

Reply:
You can check the other changes [here](https://github.com/DhruvSondhi/poliastro.github.io)

Reply:
I see on https://github.com/DhruvSondhi/poliastro.github.io/actions/runs/594940098 that this was built on several Python versions. Let's not do that - the poliastro website is not a library that will be reused, so we don't need to make sure that it runs in several Python versions.